src/Compiler/HeapReporter.cpp

```diff
@@ -48,15 +39,23 @@ void logMessage(StringRef logFilename, StringRef msg,
 #endif
 } // namespace
 
-HeapReporter::HeapReporter(
-    std::string logFilename, StringRef beforePasses, StringRef afterPasses)
+HeapReporter::HeapReporter(std::string logFilename,
+    std::vector<std::string> beforePasses, std::vector<std::string> afterPasses)
     : logFilename(logFilename) {
-  splitToSet(beforePasses, this->beforePassesSet);
-  splitToSet(afterPasses, this->afterPassesSet);
+  beforePassesSet = llvm::StringSet<>(beforePasses);
+  afterPassesSet = llvm::StringSet<>(afterPasses);
+
+  std::string beforePassesStr = "";
+  std::string afterPassesStr = "";
+
+  for (std::string &s : beforePasses)
+    beforePassesStr = beforePassesStr + s + ",";
+  for (std::string &s : afterPasses)
+    afterPassesStr = afterPassesStr + s + ",";
+
   reportBegin("onnx-mlir heap report"
               "\n--report-heap-before='" +
-              beforePasses.str() + "'\n--report-heap-after='" +
-              afterPasses.str() + "'");
+              beforePassesStr + "'\n--report-heap-after=" + afterPassesStr);
```

Comment:
can you use llvm::join() from StringExtras.h?
```
llvm::join(beforePasses, ",") + "'\n--report-heap-after=" + llvm::join(afterPasses, ",")
```

Reply:
I knew there must be some handy function for this lying around. Thanks for the pointer